### PR TITLE
Add openQueryDocument API

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
For https://github.com/microsoft/azuredatastudio/issues/16108

This should also fix https://github.com/microsoft/azuredatastudio/issues/13250

This is the same issue as the other query editor problems - setting the mode is no longer enough to result in a query editor to be opened. So to get around this I'm adding an API similar to vscode.workspace.openTextDocument to open a new query editor.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
@llali Do you have a server set up that I can test the ML changes on? Looks like it needs some special ML enabled MI server or something?

**Comment by llali** [comment]
> @llali Do you have a server set up that I can test the ML changes on? Looks like it needs some special ML enabled MI server or something?

I usually create an edge server locally. I can verify it for you if you want.

File: src/sql/workbench/api/browser/mainThreadQueryEditor.ts [hunk 1]
```diff
@@ -145,4 +148,9 @@ export class MainThreadQueryEditor extends Disposable implements MainThreadQuery
 	public $setQueryExecutionOptions(fileUri: string, options: azdata.QueryExecutionOptions): Thenable<void> {
 		return this._queryManagementService.setQueryExecutionOptions(fileUri, options);
 	}
+
+	public async $tryCreateQueryDocument(options?: { content?: string }, providerId?: string): Promise<URI> {
```
Comment (alanrenmsft) [inline src/sql/workbench/api/browser/mainThreadQueryEditor.ts:152]: why is there a `try` in the name?

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/api/browser/mainThreadQueryEditor.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/api/browser/mainThreadQueryEditor.ts:152]: Good point, I took this from the create text document stuff but there doesn't seem to be a reason for that name here

**Review by chlafreniere — APPROVED** [review]

**Comment by llali** [comment]
> > @llali Do you have a server set up that I can test the ML changes on? Looks like it needs some special ML enabled MI server or something?
> 
> I usually create an edge server locally. I can verify it for you if you want.

@Charles-Gagnon I tested ml with your branch and the bug won't repro anymore

**Review by llali — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
Thank you @llali!

**Pull request merged**